lib/SqlCall.js

```diff
@@ -16,323 +16,242 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-const iXml = require('./ixml');
-
 class SqlCall {
   /**
    * @description Creates a new SqlCall
    * @constructor
    */
   constructor() {
-    this.xml = iXml.iXmlNodeSqlOpen();
+    this.xml = '<sql>';
   }
 
   /**
    * @description adds sql query XML
    * @param {string} stmt
    * @param {object} [options]
    */
-  addQuery(stmt, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlQueryOpen(options.error) + stmt + iXml.iXmlNodeSqlQueryClose();
-    } else {
-      this.xml += iXml.iXmlNodeSqlQueryOpen() + stmt + iXml.iXmlNodeSqlQueryClose();
-    }
+  addQuery(stmt, options = {}) {
+    this.xml += `<query error='${options.error || 'fast'}'>${stmt}</query>`;
   }
 
   /**
    * @description adds sql prepare XML
    * @param {string} stmt
    * @param {object} [options]
    */
-  prepare(stmt, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlPrepareOpen(options.error) + stmt
-      + iXml.iXmlNodeSqlPrepareClose();
-    } else {
-      this.xml += iXml.iXmlNodeSqlPrepareOpen() + stmt + iXml.iXmlNodeSqlPrepareClose();
-    }
+  prepare(stmt, options = {}) {
+    this.xml += `<prepare error='${options.error || 'fast'}'>${stmt}</prepare>`;
   }
 
   /**
    * @description adds sql execute XML
    * @param {array} params
    * @param {object} [options]
    */
-  execute(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlExecuteOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlExecuteOpen();
-    }
+  execute(params, options = {}) {
+    this.xml += `<execute error='${options.error || 'fast'}'>`;
 
     if (params && params.length) {
       for (let i = 0; i < params.length; i += 1) {
-        if (params[i].length > 1) {
-          this.xml += iXml.iXmlNodeSqlParmOpen(params[i][1]) + params[i][0]
-          + iXml.iXmlNodeSqlParmClose();
-        } else {
-          this.xml += iXml.iXmlNodeSqlParmOpen() + params[i][0] + iXml.iXmlNodeSqlParmClose();
-        }
+        this.xml += `<parm${params[i].length ? ` io='${params[i][1]}'` : ''}>${params[i][0]}</parm>`;
       }
     }
-    this.xml += iXml.iXmlNodeSqlExecuteClose();
+    this.xml += '</execute>';
   }
 
   /**
    * @description adds sql table XML
    * @param {array} params
    * @param {object} [options]
    */
-  tables(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlTablesOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlTablesOpen();
-    }
-
+  tables(params, options = {}) {
+    this.xml += `<tables${options.error ? ` error='${options.error}'` : ''}>`;
```

Comment:
```suggestion
    this.xml += `<tables error='${options.error || 'fast'}'>`;
```
It's a slight behavior change, but much clearer.

Reply:
And by behavior change, I mean that we would be extraneously sending the default error option instead of leaving it blank and letting XMLSERVICE default it. The default is still the same, however, so the only difference is that we send some extra data in the XML that we didn't _need_ to, but it's unlikely to hurt anything.

Reply:
Makes sense we would need to update the unit tests as well.